repos/system_upgrade/common/libraries/overlaygen.py

```diff
@@ -68,6 +68,21 @@
 this constant.
 """
 
+_MAX_DISK_IMAGE_SIZE_MB = 1024**2  # 1*TB
+"""
+Maximum size of the created (sparse) images.
+
+Defaults to 1TB. If a disk with capacity larger than _MAX_DISK_IMAGE_SIZE_MB
+is mounted on the system, the corresponding image used to store overlay
+modifications will be capped to _MAX_DISK_IMAGE_SIZE_MB.
+
+Engineering rationale:
+   This constant was introduced to prevent leapp from creating files that are
+   virtually larger than the maximum file size supported by the file system.
+   E.g. if the source system hosts /var/lib/leapp on EXT4, then we cannot
+   create a file larger than 16TB.
```

Comment:
I would extend the reasoning why we chose 1TB.

```suggestion
   This constant was introduced to prevent leapp from creating files that are
   virtually larger than the maximum file size supported by the file system.
   E.g. if the source system hosts /var/lib/leapp on EXT4, then we cannot
   create a file larger than 16TB.
   We create these "disk images" to be able to verify the system has enough
   disk space to perform the RPM upgrade transaction. From our experience,
   we are not aware of any system which could have installed so much content
   by RPMs, that we would need 1TB of the free space on a single FS. So we consider
   this cap to be safe enough and it does not cost us too much.
```